generate-eslint-resolved-configs.js

```diff
@@ -0,0 +1,115 @@
+import { exec } from 'node:child_process';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+async function checkPathExists(p) {
+  try {
+    await fs.access(p);
+
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+function sortObjectKeysAlphabetically(obj) {
+  const entries = Object.entries(obj);
+
+  entries.sort((a, b) => a[0].localeCompare(b[0]));
+
+  return Object.fromEntries(entries);
+}
+
+function sortArrayAlphabetically(array) {
+  const clonedArray = [...array];
+  clonedArray.sort((a, b) => a.localeCompare(b));
+
+  return clonedArray;
+}
+
+const NUMBER_TO_SEVERITY_MAP = { 0: 'off', 1: 'warn', 2: 'error' };
+
+function preprocessConfig(configContent) {
+  const config = JSON.parse(configContent);
+
+  config.languageOptions.globals = sortObjectKeysAlphabetically(
+    config.languageOptions.globals
+  );
+
+  config.plugins = sortArrayAlphabetically(config.plugins);
+
+  const rulesEntries = Object.entries(config.rules);
+
+  rulesEntries.sort((a, b) => a[0].localeCompare(b[0]));
+
+  for (const rule of rulesEntries) {
+    const [ruleName, ruleOptions] = rule;
+
+    if (Array.isArray(ruleOptions)) {
+      const severity = ruleOptions[0];
+
+      ruleOptions[0] = NUMBER_TO_SEVERITY_MAP[severity];
+    } else {
+      // This isn't supposed to happen
+      throw new Error(
+        `Rule Options must always be an array, but this is not the case for: ${ruleName}`
+      );
+    }
+  }
+
+  config.rules = Object.fromEntries(rulesEntries);
+
+  return JSON.stringify(config, null, 2);
+}
+
+async function getResolvedConfig(filepath) {
+  return new Promise((resolve, reject) => {
+    exec(`npx eslint --print-config ${filepath}`, (err, stdout, stderr) => {
```

Comment:
test if this does what you think it does.

Reply:
I read the documentation and tested it and it works as I thought

https://nodejs.org/api/child_process.html#child_processexeccommand-options-callback